datastream-jdbc-connector/src/main/java/com/linkedin/datastream/connectors/jdbc/JDBCConnectorTask.java

```diff
@@ -105,7 +105,7 @@ private void processResults(ResultSet resultSet) throws SQLException, IOExceptio
 
             HashMap<String, String> meta = new HashMap<>();
             meta.put(BrooklinEnvelopeMetadataConstants.EVENT_TIMESTAMP, String.valueOf(System.currentTimeMillis()));
-            BrooklinEnvelope envelope = new BrooklinEnvelope(null, record, null, meta);
+            BrooklinEnvelope envelope = new BrooklinEnvelope(checkpoint, record, null, meta);
```

Comment:
This will work.

But what if, we pass properly populated GenericRecord instead of checkpoint as Long? That might be a generic solution from Transport Provider code's point of view.